packages/apps/human-app/server/src/modules/jobs-discovery/jobs-discovery.service.ts

```diff
@@ -0,0 +1,47 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { lastValueFrom } from 'rxjs';
+import {
+  JobsDiscoveryParamsCommand,
+  JobsDiscoveryParamsData,
+  JobsDiscoveryResponse,
+} from './interfaces/jobs-discovery.interface';
+import { HttpService } from '@nestjs/axios';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+
+@Injectable()
+export class JobsDiscoveryService {
+  constructor(
+    public httpService: HttpService,
+    @InjectMapper() private readonly mapper: Mapper,
+  ) {}
+
+  async processJobsDiscovery(
+    url: string,
+    jobsDiscoveryParamsCommand: JobsDiscoveryParamsCommand,
+  ): Promise<JobsDiscoveryResponse> {
+    const jobsDiscoveryParamsData = this.mapper.map(
+      jobsDiscoveryParamsCommand,
+      JobsDiscoveryParamsCommand,
+      JobsDiscoveryParamsData,
+    );
+    try {
+      const options = {
+        method: 'GET',
+        url: `${url}/jobs`,
+        params: jobsDiscoveryParamsData,
+      };
+      const response = await lastValueFrom(this.httpService.request(options));
```

Comment:
what is the motivation behind lastValueFrom

Reply:
HttpService methods is an Observable -> [docs](https://docs.nestjs.com/techniques/http-module)